docs/20_smart-contracts/10_getting-started/10_dune-guide/index.md

```diff
@@ -0,0 +1,634 @@
+-- Setting up CDT Wasm Toolchain 3.1.0 at /usr
+-- Setting up CDT Wasm Toolchain 3.1.0 at /usr
+-- The C compiler identification is Clang 9.0.1
+-- The CXX compiler identification is Clang 9.0.1
+-- Detecting C compiler ABI info
+-- Detecting C compiler ABI info - failed
+-- Detecting C compile features
+-- Detecting C compile features - done
+-- Detecting CXX compiler ABI info
+-- Detecting CXX compiler ABI info - failed
+-- Detecting CXX compile features
+-- Detecting CXX compile features - done
+-- Configuring done
+-- Generating done
+-- Build files have been written to: /host/home/lupaxel/work/contracts/hello/build/hello
+[ 66%] Performing build step for 'hello_project'
+Scanning dependencies of target hello
+[ 50%] Building CXX object CMakeFiles/hello.dir/hello.obj
+Warning, empty ricardian clause file
+[100%] Linking CXX executable hello.wasm
+[100%] Built target hello
+[ 77%] No install step for 'hello_project'
+[ 88%] No test step for 'hello_project'
+[100%] Completed 'hello_project'
+[100%] Built target hello_project
+```
+
+Now you will deploy the contract.
+
+### Step 4 - Deploy Contract
+
+Run the following command to deploy your `hello` contract:
+
+```shell
+dune --deploy build/hello hello
+```
+
+The command will deploy the compiled contract residing at `build/hello` to account `hello`:
+
+```
+#         eosio <= eosio::setcode               {"account":"hello","vmtype":0,"vmversion":0,"code":"0061736d0100000001410d60017e0060017f0060027f7f00...
+#         eosio <= eosio::setabi                {"account":"hello","abi":"0e656f73696f3a3a6162692f312e3200010268690001026e6d046e616d6501000000000000...
+```
+
+You can check that the contract was indeed deployed by running the following command:
+
+```shell
+dune -- cleos get code hello
+```
+
+The command sends the name of code account, `hello`, where the contract was deployed, to the `get_code` endpoint, which replies with the hash of the deployed contract:
+
+```
+code hash: 2737de70263c4ac5ac8e04760e5d7426a8d7c9a1680f2bcc3e44354185badc34
+```
+
+If the code hash is non-zero, as above, then the contract was deployed successfully. You can now test your smart contract.
+
+## Smart Contract Testing
+
+You can use DUNE to start testing your smart contract locally first. Later on, you should test your contract on an EOS testnet, and eventually on the real EOS mainnet.
+
+To test your smart contract, you typically send actions to it. Then you inspect the endpoint response to those actions and their effect on the blockchain state, if applicable.
+
+### Send Actions
+
+In the `hello` contract, there is only one `world` action that receives an EOS name as a parameter. To test the contract, you need to send the `world` action to your contract with an appropriate argument.
+
+To that end, specify the `hello` account and send the action `world` with your name ("john" in the example below) as argument - note that you must use lowercase and abide by the EOS account name rules:
```

Comment:
Let's remove this line. It doesn't add much that isn't already in the code below, and this guide comes before the account guide which talks about EOS account name rules, so it's putting the cart before the horse

Reply:
Done